### PR TITLE
Demo app revamp

### PR DESCRIPTION
**Pull request by iainmcgin**
- Removes all references to Google from the app
- Focuses the app on the single IDP use case
- Demonstrates persisting authorization state to shared prefs
- Enables Java 8 features, simplifies some asynchronous code

---
This change is [Reviewable](https://reviewable.io/reviews/openid/appauth-android/185)

### CODE REVIEW

**Comment by iainmcgin** [comment]
Context: http://lists.openid.net/pipermail/openid-specs-ab/Week-of-Mon-20170306/006456.html

File: app/README-Google.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Configuring Google Sign In
```
Comment (WilliamDenniss) [inline app/README-Google.md]: Google Sign In and OAuth?

File: app/README.md [hunk 2]
```diff
@@ -1,151 +1,54 @@
...
-------------------------------
+# AppAuth for Android - Demo App
 
 This app demonstrates the AppAuth library by performing an authorization code
-flow with Google as the authorization service. While the app compiles without
-modification, some configuration is required in order to see Google as an
-authorization option
+flow with an authorization service. The configuration contained in `res/raw/idp_config.json`
+must be modified in order for the app to function. Warnings are supplied when the app is run
+if the configuration is invalid.
```
Comment (WilliamDenniss) [inline app/README.md]: "with an invalid configuration"

File: app/README.md [hunk 3]
```diff
@@ -1,151 +1,54 @@
...
+The configuration file MUST contain a JSON object. The following properties can be specified:
 
-The configuration properties that must be modified can be found in
-[res/values/idp_configs.xml](res/values/idp_configs.xml).
-First, An OAuth2 client ID for Google Sign In must be created.
-The [quick-start configurator](https://goo.gl/pl2Fu2) can be used to
-generate this, or it can be done directly on the
-[Google Developer Console](https://console.developers.google.com/apis/credentials?project=_).
+  - `redirect_uri` (required): The redirect URI to use for receiving the authorization response.
+    This can either be a custom scheme URI (example:/path) or an https app link
```
Comment (WilliamDenniss) [inline app/README.md]: `com.example.app:/oauth2redirect/example-provider` is the example I use in the [BCP](https://tools.ietf.org/html/draft-ietf-oauth-native-apps-08#section-7.1). I know it was just an example, but technically `example:/path` would violate IANA requirements for private-use URI schemes (not domain-based).

File: app/README-Google.md (hunk 1 above)
Comment (iainmcgin) [inline app/README-Google.md]: I'll adopt the title you used in your readme, "Using AppAuth for Android with Google".

File: app/README.md (hunk 2 above)
Comment (iainmcgin) [inline app/README.md]: Done.

File: app/README.md (hunk 3 above)
Comment (iainmcgin) [inline app/README.md]: Done.

**Comment by xerners** [comment]
👍 @WilliamDenniss @iainmcgin - We will be doing our bit shortly for review.

**Pull request merged**